src/Pool.sol

```diff
@@ -95,15 +112,37 @@ contract Pool is Ownable {
     //
     function setAddresses(address _chickenBondManagerAddress) external onlyOwner {
         chickenBondManagerAddress = _chickenBondManagerAddress;
-        renounceOwnership();
     }
 
     fallback() external payable {}
     receive() external payable {}
 
-    function updatePool(address _storageProvider, uint256 amount) public {
-        lockedCapital[_storageProvider] += amount / 2;
-        totalWorkingCapital += amount / 2;
+    function harvestRewards() public {
+        updatePool(address(0), 0);
+
+        PoolStaker storage staker = poolStakers[msg.sender];
+        uint256 rewards = staker.amount * accumulatedRewardsPerShare / REWARDS_PRECISION;
+        uint256 rewardsToHarvest = rewards - staker.rewardDebt;
+
+        staker.rewardDebt = rewards;
+
+        if (rewardsToHarvest == 0) {
+            return;
+        }
+
+        emit HarvestRewards(msg.sender, rewardsToHarvest);
+        // payable(manager).transfer(rewardsToHarvest);
```

Comment:
why is this comment here ?

Reply:
I had to split this into multiple PR's, otherwise it would be way too many changes all at once, and that's a part of the next PR I'm going to open

Reply:
Also, I'm still not sure if we are going to transfer the rewards back to the manager every single time there's an interaction with the Pool. We might only need do it when there's a `chickenIn` or `chickenOut`, or someone explicitly calls a function to get rewards back to the manager.

This is still in flux so that's why I have the comment there

Reply:
I had the idea that this would be manager agnostic.

Reply:
It can be. This will be replaced with a call to `withdraw` that receives the address where the funds are being sent